deployment.md

```diff
@@ -0,0 +1,162 @@
+Therefore, you need to specify the owner address when deploying the factory. You should store the owner address to `$FACTORY_OWNER_ADDRESS` and private key for further usage to `$FACTORY_OWNER_PRIVATE_KEY`.
+You can deploy the factory with any account, so specify its private key in `$DEPLOYER_PRIVATE_KEY`.
+
+To deploy the Oracle factory, run the following command:
+
+```shell
+forge create --optimizer-runs 100 -via-ir src/factory/ERC20OracleOnlyFactory.sol:ERC20OracleOnlyFactory --constructor-args $FACTORY_OWNER_ADDRESS --private-key $DEPLOYER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed factory, that you should save to `$ORACLE_FACTORY_ADDRESS`.
+
+---
+
+To deploy the Paymaster factory, run the following command:
+
+```shell
+forge create --optimizer-runs 100 -via-ir src/factory/ERC20PaymasterOnlyFactory.sol:ERC20PaymasterOnlyFactory --constructor-args $FACTORY_OWNER_ADDRESS --private-key $DEPLOYER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed factory, that you should save to `$PAYMASTER_FACTORY_ADDRESS`.
+
+## Deployment
+
+Below commands require some parameters and environment variables:
+
+- <salt> - the random bytes32 value that is used to derive the address of the deployed contract. It can be unique for each deployment.
+- <chain_id> - the chain id of the network where the contract will be deployed.
+- `$NATIVE_POOL_ADDRESS` - the address of the Token <> ETH liquidity pool.
+- `$POOL_NATIVE_TOKEN_ADDRESS` - the address of the native token FROM THE LIQUIDITY POOL.
+- `$TOKEN_ADDRESS` - the address of the token that the paymaster will operate with.
+- `$PAYMASTER_OWNER_ADDRESS` - the address of the paymaster owner, that will be able to change the price markup.
+- <price_markup> - the price markup, where 1000000 means 100%. It is used to calculate the price of the token in the paymaster. For example, 100% markup and that the user will pay at the actual price, and 120% means the user pays 20% more. Possible values are from 0 to 4294967295.
+- <max_price_markup> - the maximum price markup that the paymaster owner can set.
+
+To deploy FixedOracle: run:
+
+```shell
+cast send $ORACLE_FACTORY_ADDRESS "deployFixedOracle(bytes32 salt, int256 _price)" <salt> 100000000 --private-key $FACTORY_OWNER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed oracle, that you should save to `$FIXED_ORACLE_ADDRESS`.
+
+---
+
+To deploy TwapOracle: run:
+
+```shell
+cast send $ORACLE_FACTORY_ADDRESS "deployTwapOracle(bytes32,address,uint32,address)" <salt> $NATIVE_POOL_ADDRESS 61 POOL_NATIVE_TOKEN_ADDRESS --private-key $FACTORY_OWNER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed oracle, that you should save to `$TWAP_ORACLE_ADDRESS`.
```

Comment:
Consider replacing "save to" with "store in".

"Store in" is more precise when referring to placing a value inside a variable or a memory location. "Save to" can be ambiguous and might suggest saving to an external system, whereas "store in" makes it clear that the value is being assigned to a variable.
```suggestion
The command will return the address of the deployed oracle, that you should store in `$FIXED_ORACLE_ADDRESS`.

---

To deploy TwapOracle: run:

```shell
cast send $ORACLE_FACTORY_ADDRESS "deployTwapOracle(bytes32,address,uint32,address)" <salt> $NATIVE_POOL_ADDRESS 61 POOL_NATIVE_TOKEN_ADDRESS --private-key $FACTORY_OWNER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
```

The command will return the address of the deployed oracle, that you should store in `$TWAP_ORACLE_ADDRESS`.
```

Reply:
60879dbb5b1bb13b48bb5bc131b6e990d3862a88